cloudfour/cloudfour.com-patterns
# Multi-watch

## Overview

Prior to this PR, file preprocessing handled by Gulp (SVG → Twig and Theo → MDX) happened only on startup and never again.

This PR changes that so file changes will be seen and take effect immediately.

What I _didn't_ want to do is pipe Storybook through Gulp. My experience is that any Webpack projects piped through Gulp just end up being a single build step, and that seems like a waste of the engineering.

Instead, I used a two-part solution:

1. I've added a new `watch` task in Gulp. This is very simple, it just watches some file globs and runs the appropriate tasks.
1. I then added the popular [npm-run-all](https://github.com/mysticatea/npm-run-all), which lets us run npm scripts in parallel.

As a bonus, npm-run-all lets us simplify a few of our script definitions.

The main issue with this approach is that the CLI output is a bit messy, with Gulp output appended to Webkit output. If this bothers us _too_ much, we could consider making the Gulp output `--silent` when run in parallel. But I think it's a fair compromise.

## Testing

1. Check out branch.
1. Run `npm run storybook`
1. Observe that local server launches correctly.
1. Try changing a design token, for example updating a color value.
1. Observe that the documentation page updates as expected.
1. Try changing the Viewbox size of `src/assets/icons/heart.svg`.
1. Observe that the icon object docs update accordingly.
